cancat/j1939stack.py

```diff
@@ -77,7 +78,54 @@ def parseArbid(arbid):
     edp = (prioPlus >> 1) & 1
     dp = prioPlus & 1
 
-    return prio, edp, dp, pf, ps, sa
+    return arbid, prio, edp, dp, pf, ps, sa
+
+
+def getArbtupInfo(arbtup):
```

Comment:
what do you think about creating a `arbtup` cache within CanCat/J1939?
we keep parsing out the whole thing for every message...
what about 
```
J1939Interface._arbcache = {}
...
def getArbData(arbid):
    if arbtup in self._arbcache:
        return self._arbcache.get(arbid)
    ...
```

Reply:
I'm not sure... the PGN for TP messages may match the PGN used in some single-frame messages, so it feels like it may be misleading unless we also include length in the cache lookup?

Perhaps we should cache the original CAN messages and each J1939 parsed message could have a reference to the original CAN message frame (or frames) it came from?  Then we could get the arbid from the referenced CAN frame.  But that feels like it would completely change the design of what you've done and I'm not sure such a big change is worth it.